nautobot/extras/jobs.py

```diff
@@ -378,6 +379,22 @@ def deserialize_data(cls, data):
 
         return return_data
 
+    def validate_data(self, data):
+        vars = self._get_vars()
+
+        if type(data) is not dict:
+            raise ValidationError("Job data needs to be a dict!")
+
+        for k, v in data.items():
+            if k not in vars:
+                raise ValidationError({k: "Job data contained an unknown property"})
+
+            # defer validation to the form field object
+            try:
+                vars[k].as_field().clean(v)
+            except ValidationError as e:
+                raise ValidationError({k: e.message})
+
```

Comment:
Seems like we might also need a check here to make sure that `data` isn't missing any `required=True` vars?

Would something like `self.as_form(data=data).clean()` work instead?